_source/_posts/2025-02-24-okta-hosted-sign-in-widget.md

```diff
@@ -0,0 +1,776 @@
+---
+layout: blog_post
+title: "From Meh to Wow: Customize Your Okta Sign-In Experience"
+author: nesh-popovic
+by: internal-contributor
+communities: [javascript]
+description: "Tutorial on customizing your Okta Hosted Sign-In Widget to the next level."
+tags: [sign-in-custom-domain, custom-domain, okta-hosted-sign-in]
+tweets:
+- ""
+- ""
+- ""
+image: blog/okta-hosted-sign-in-widget/sign-in-widget-social-image.jpeg
+type: awareness
+---
+The Okta Sign-In Widget is a powerful tool that allows developers to integrate Okta authentication into their web applications seamlessly. But did you know you can also customize the look and feel of the widget to match your application(s)' design, even if the widget is hosted on Okta? Whether you want to modify colors, add custom branding, or adjust the layout, Okta provides flexibility to make the widget fit your brand's identity.
+
+In this guide, we'll transform the Okta-hosted widget using standard Javascript and CSS and its built-in functionality.
+
+{:toc}
+
+**Before**
+{% img blog/okta-hosted-sign-in-widget/okta-hosted-sign-in-widget-before.gif alt:"animated gif of default static Okta hosted sign-in widgets" %}{: .center-image }
+
+Our goal:
+
+**After**
+{% img blog/okta-hosted-sign-in-widget/okta-hosted-sign-in-widget-after.gif  alt:"animated gif of customized Okta hosted sign-in widget with sliding window effect for new signups" width:"800" %}{: .center-image }
+
+
+## Okta Sign-In Widget basics
+Before diving into customization, let's go over a few basics. "Okta-hosted widget" means that when the user needs to authenticate, you redirect them from your application to Okta, and the login screen that the user is presented with is hosted by Okta (it could be on a custom domain). Hence, all the widget code is on Okta itself (and not in the application), which makes it very convenient for your developers.
+
+**Prerequisites to this tutorial**
+This guide assumes you already have a Single Sign-On (SSO) application that redirects to Okta for authentication. In addition, you already have a customized Okta-hosted widget with a [custom domain](https://help.okta.com/en-us/content/topics/settings/settings-configure-custom-url.htm) enabled to make the necessary edits I will be demonstrating in this tutorial. This tutorial doesn't use the Third-generation widget but uses the Sign-In Widget **version ^7** instead. And lastly, this guide assumes you have [Self-Service Registration (SSR)](https://support.okta.com/help/s/article/sign-up-link-missing-from-okta-login-page-during-sp-initiated-login?language=en_US) also enabled to handle user self-registration.  
```

Comment:
```suggestion
This guide assumes you already have a Single Sign-On (SSO) application that redirects to Okta for authentication. In addition, you already have a customized Okta-hosted widget with a [custom domain](https://help.okta.com/en-us/content/topics/settings/settings-configure-custom-url.htm) enabled to make the necessary edits I will be demonstrating in this tutorial. This tutorial uses the Sign-In Widget **version ^7**. And lastly, this guide assumes you have [Self-Service Registration (SSR)](https://support.okta.com/help/s/article/sign-up-link-missing-from-okta-login-page-during-sp-initiated-login?language=en_US) also enabled to handle user self-registration.  
```

Highly recommend splitting this out into bullet points or something easy to scan for prereqs